internal/pkg/cli/svc_package.go

```diff
@@ -470,5 +488,9 @@ func buildSvcPackageCmd() *cobra.Command {
 	cmd.Flags().StringVar(&vars.tag, imageTagFlag, "", imageTagFlagDescription)
 	cmd.Flags().StringVar(&vars.outputDir, stackOutputDirFlag, "", stackOutputDirFlagDescription)
 	cmd.Flags().BoolVar(&vars.uploadAssets, uploadAssetsFlag, false, uploadAssetsFlagDescription)
+	cmd.Flags().BoolVar(&vars.showDiff, diffFlag, false, diffFlagDescription)
+
+	cmd.MarkFlagsMutuallyExclusive(diffFlag, stackOutputDirFlag)
+	cmd.MarkFlagsMutuallyExclusive(diffFlag, uploadAssetsFlag)
```

Comment:
Why do we not allow them to specify upload assets flag?

Reply:
`copilot svc package --diff --upload-assets`  - not sure if it makes sense to upload assets while they just want to show diff 🤔  

We upload env file, custom resource and addons with `--upload-assets`. Env files URLs are passed via. `Parameters` iirc, custom resources/addons URLs are deterministic and are rendered into the template regardless whether they are uploaded. Therefore, with/without `--upload-assets` should render the same template. We can definitely relax this constraint now or later, I am not feeling strong on this.